client.go

```diff
@@ -119,6 +119,49 @@ func (c *Client) Get(ctx context.Context, path string, options map[string]string
 	return resp.Body, nil
 }
 
+func (c *Client) Put(ctx context.Context, path string, body interface{}) (io.Reader, error) {
+	u, err := url.Parse(c.Base.String())
+	if err != nil {
+		return nil, fmt.Errorf("bad path: %w", err)
+	}
+
+	u.Path = path
+
+	b, err := json.Marshal(body)
+	if err != nil {
+		return nil, fmt.Errorf("could not marshal body: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, u.String(), bytes.NewReader(b))
+	if err != nil {
+		return nil, fmt.Errorf("could not create request: %w", err)
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := c.HTTP.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("request (%+v) failed: %w", req, err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		var buf bytes.Buffer
+		tee := io.TeeReader(resp.Body, &buf)
+		errResp := ErrorResponse{}
+		if err := json.NewDecoder(tee).Decode(&errResp); err != nil {
+			return nil, fmt.Errorf("could not decode error response %s: %w", buf.String(), err)
+		}
+
+		// log.Printf("%s -> %+v", buf.String(), errResp)
+		if errResp.Error() != "" {
+			return nil, fmt.Errorf("%s: %s", resp.Status, errResp.Error())
+		}
+
+		return nil, fmt.Errorf("%s", resp.Status)
+	}
+
+	return resp.Body, nil
```

Comment:
In other circumstances I might have read the body and put in a byte array or string, but I think this is fine.